components/formats-gpl/src/loci/formats/in/FlexReader.java

```diff
@@ -1234,8 +1235,14 @@ private void groupFiles(String[] fileList, MetadataStore store)
         firstIFD = parser.getFirstIFD();
         ifdCount = parser.getIFDOffsets().length;
       }
-      boolean compressed =
-        firstIFD.getCompression() != TiffCompression.UNCOMPRESSED;
+      Boolean compressed = null;
+      try {
+        compressed =
+          firstIFD.getCompression() != TiffCompression.UNCOMPRESSED;
+      }
+      catch (EnumException e) {
+        LOGGER.trace("Could not get compression for " + file, e);
```

Comment:
In practice the only two types of compressions we might have ever encountered with the Opera Flex format are UNCOMPRESSED and LURAWAVE, right?

Should `compressed` be set to True in that case to capture the fact the data is compressed and ensure minimal changes with regard to the compression comparison below?

Reply:
`curated/flex/karsten/Columbus_ExtendedData/Images/AFo_U2OS_20xW_070209/Meas_01(2007-02-09_15-36-40)` (which we do not test) has undefined compression type 34692, but as far as I can tell the rest of our sample data is either uncompressed (type 0 or 1) or Lurawave (type 65535).

1f27a1a8a1 sets `compressed` to `true` here, and also removes a `continue` later on so that attempting to initialize a Lurawave-compressed file will fail with a more informative message. Lurawave-compressed files will throw an exception during `setId` now; with 6.14.0 and no Lurawave jar, the exception was thrown in `openBytes`. Since we do not support Lurawave compression at all anymore, I think throwing an exception during initialization is better, particularly as it will prevent importing unsupported data into OMERO.